modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaEntityBrowserWidgetTest.php

```diff
@@ -0,0 +1,206 @@
+      ->setComponent('featured_media_field', $view_display_options);
+    $display->save();
+
+    $this->drupalLogin($this->drupalCreateUser([], '', TRUE));
+  }
+
+  /**
+   * Tests the featured media entity browser widget.
+   */
+  public function testFeaturedMediaEntityBrowserWidget(): void {
+    $this->drupalGet('node/add/page');
+
+    // Assert that all the entity browser elements are displayed.
+    $this->assertSession()->buttonExists('Select images');
+    $this->assertSession()->fieldExists('Caption');
+    $this->assertSession()->pageTextContains('The caption that goes with the referenced media.');
+    $this->assertSession()->buttonExists('Add another item');
+
+    // Select one media image from the entity browser.
+    $this->getSession()->getPage()->pressButton('Select images');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->switchToIFrame('entity_browser_iframe_test_images');
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:1]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Assert the image was selected and the widget shows the proper buttons.
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->buttonNotExists('Select images');
+    $this->assertSession()->buttonExists('Remove');
+    $this->assertSession()->buttonExists('Edit');
+
+    // Add the other media image item.
+    $this->getSession()->getPage()->pressButton('Add another item');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->getPage()->pressButton('Select images');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->switchToIFrame('entity_browser_iframe_test_images');
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:2]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertSession()->buttonNotExists('Select images');
+
+    // Fill in the other fields and save the node.
+    $this->getSession()->getPage()->fillField('featured_media_field[0][caption]', 'Image 1 caption');
+    $this->getSession()->getPage()->fillField('featured_media_field[1][caption]', 'Image 2 caption');
+    $this->getSession()->getPage()->fillField('Title', 'Test entity browser widget');
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert that the values were saved.
+    $this->assertSession()->pageTextContains('Featured media field');
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertSession()->pageTextContains('Image 1 caption');
+    $this->assertSession()->pageTextContains('Image 2 caption');
+
+    // Edit the node to remove the first media.
```

Comment:
Before removing, reorder the fields and see if they got reordered (you can load the node and assert correct deltas). See `TableDragTest` for way to table drag.

Then remove one, including the caption ,and assert it got removed completely also after save.